src/argowrapper/engine/argo_engine.py

```diff
@@ -86,18 +88,52 @@ def _get_workflow_log_dict(self, workflow_name: str) -> Dict:
             _check_return_type=False,
         ).to_dict()
 
-    def _get_log_errors(self, status_nodes_dict: Dict) -> List[Dict]:
+    def _get_workflow_phase(self, workflow_name: str) -> str:
+        phase_return = self.api_instance.get_workflow(
+            namespace=ARGO_NAMESPACE,
+            name=workflow_name,
+            fields="status.phase",
+            _check_return_type=False,    
+        ).to_dict()
+        return phase_return["status"].get("phase")
+    
+    def _get_workflow_node_artifact(self, uid: str, node_id: str) -> str:
+        return self.artifact_api_instance.get_output_artifact_by_uid(
+            uid=uid,
+            node_id=node_id,
+            artifact_name="main-logs",
+            _check_return_type=False,
+        ).read().decode()
+
+    def _get_log_errors(self, uid: str, status_nodes_dict: Dict) -> List[Dict]:
         errors = []
-        for _, step in status_nodes_dict.items():
-            if step.get("phase") in ("Failed", "Error"):
+        for pod_id, step in status_nodes_dict.items():
```

Comment:
is `node_id` more accurate? See also comment below.

Reply:
Updated